src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -137,7 +137,7 @@ public function getTypes($class, $property, array $context = array())
             return;
         }
 
-        if (!in_array($prefix, ReflectionExtractor::$arrayMutatorPrefixes)) {
+        if (!in_array($prefix, ReflectionExtractor::$defaultArrayMutatorPrefixes)) {
```

Comment:
I mean your edit yes: if ReflectionExtractor needs this, why shouldn't PhpDocExtractor?
But note that I'd suggest to wait for some feedback from @dunglas before spending more time on code here, just to not waste yours :)